### ISSUE TITLE
tailwindcss/enforce-canonical does not report break-words -> wrap-break-word

### ISSUE THREAD

**Issue by yashau** [sergioazoc/oxlint-tailwindcss#16]
## Description

`tailwindcss/enforce-canonical` does not report `break-words`, even though Tailwind CSS v4 canonicalizes it to `wrap-break-word`.

## Versions

- `oxlint`: `1.63.0`
- `oxlint-tailwindcss`: `0.7.0`
- `tailwindcss`: `4.2.4`
- OS: Windows

## Config

```jsonc
{
  "jsPlugins": ["oxlint-tailwindcss"],
  "settings": {
    "tailwindcss": {
      "entryPoint": "src/index.css",
      "rootFontSize": 16
    }
  },
  "rules": {
    "tailwindcss/enforce-canonical": "error"
  }
}
```

`src/index.css`:

```css
@import "tailwindcss";
```

## Repro

```tsx
export function Example() {
  return (
    <div className="absolute whitespace-pre-wrap break-words overflow-hidden">
      text
    </div>
  );
}
```

Run:

```bash
DEBUG=oxlint-tailwindcss pnpm exec oxlint --threads=1 -D tailwindcss/enforce-canonical src/components/PdfPage/annotations/CommentLayer.tsx
```

## Actual Result

No diagnostic is reported:

```text
Found 0 warnings and 0 errors.
[oxlint-tailwindcss] Loaded design system from ".../src/index.css"
[oxlint-tailwindcss] src/components/PdfPage/annotations/CommentLayer.tsx -> src/index.css
```

## Expected Result

The rule should report something like:

```text
"break-words" can be written as "wrap-break-word". Use the canonical form.
```

## Notes

This does not appear to be an entry-point/configuration issue. Debug logging shows the plugin successfully loads the Tailwind design system from `src/index.css`.

Tailwind CSS v4 itself appears to know this canonical mapping: `break-words` -> `wrap-break-word`.

**Comment by sergioazoc** [comment]
Confirmed and fixed on `main` (will ship in v0.7.1).

## Root cause

`enforce-canonical` resolves named classes via a precomputed `canonicalMap` built from `ds.getClassList()`. But `getClassList()` does not enumerate legacy v3 spellings, even when they still produce valid CSS in v4 and `ds.canonicalizeCandidates()` rewrites them. `break-words` is one of those — it's omitted from the class list, so the rule never saw a canonical form to suggest.

I probed the API systematically and found **~280 legacy classes** in the same shape, falling into these groups:

- Fixed renames: `break-words` → `wrap-break-word`, `order-none` → `order-0`, `overflow-ellipsis` → `text-ellipsis`, `flex-grow{,-0,-1}` → `grow{,-0,-1}` (`flex-grow-1` aliases to `grow`, the v4 default), `flex-shrink{,-0,-1}` → `shrink*`, `decoration-clone/slice` → `box-decoration-clone/slice`, `bg-gradient-to-{8 dirs}` → `bg-linear-to-*`.
- Pattern renames: `start-N`/`end-N` (and negatives, including fractionals like `start-1/2`) → `inset-s-N`/`inset-e-N`.

## Fix

The precompute step in `sync-loader.ts` now feeds a curated seed list of legacy spellings (plus dynamically derived `start-*`/`end-*` from the `inset-{s,e}-*` utilities present in the design system) into `canonicalizeCandidates()`, and adds the diffs to the canonical map. Legacy classes are also appended to `validClasses` so `no-unknown-classes` doesn't flag them either.

Arbitrary forms (`flex-grow-[2]`, `start-[10px]`) keep going through the worker, which canonicalizes both the legacy prefix and the value in a single pass — e.g. `start-[10px]` → `inset-s-2.5` (10px = 0.625rem = spacing token 2.5).

Disk cache version bumped (11 → 13) so existing caches are invalidated automatically.

## Repro test

Your example is now part of the regression suite:

```tsx
<div className="absolute whitespace-pre-wrap break-words overflow-hidden" />
//                                       ^^^^^^^^^^^
// "break-words" can be written as "wrap-break-word". Use the canonical form.
```

Thanks for the report and the clean repro!
